app/models/comment.php

```diff
@@ -22,6 +24,146 @@ class Comment extends AppModel
         ),
       ),
   );
+
+  public static function countAll($thread_id)
+  {
+    $db = DB::conn();
+    return (int) $db->value("SELECT COUNT(*) FROM comment WHERE 
+                                  thread_id = ? ", array($thread_id));
+  }
+
+  public static function getAll($offset, $limit, $thread_id)
+  {
+    $comments = array();
+    $db = DB::conn();
+    $rows = $db->rows("SELECT * FROM comment 
+                        WHERE thread_id = ? 
+                        ORDER BY created ASC LIMIT {$offset}, {$limit}", 
+                        array($thread_id));
+  
+   foreach ($rows as $row) {
+      $comments[] = new self($row);
+    }
+    return $comments;
+  }
+
+  public static function get($id)
+  {
+      $db = DB::conn();
+      $row = $db->row('SELECT * FROM comment WHERE id = ?', array($id));
+
+      if (!$row) {
+          throw new RecordNotFoundException('no record found');
+      }
+      return new self($row);
+  }
+
+  public function delete()
+  {
+    try {
+      $db = DB::conn();
+      $db->begin();
+      $params = array(
+          $this->id, 
+          $_SESSION['user_id']
+      );
+      $db->query('DELETE FROM comment WHERE id = ? && user_id = ?', $params );
+      $db->commit();
+    } catch (Exception $e) {
+      $db->rollback();
+    }
+  }
+
+  public function isUserComment()
+  {
+    return $this->user_id === $_SESSION['user_id'];
+  }
+
+  public function addLike()
+  {
+    try {
+      $db = DB::conn();
+      $db->begin();
+      $params = array(
+          'comment_id' => $this->id, 
+          'user_id' => $_SESSION['user_id'] 
+      );
+      $db->insert('likes', $params);
+      $db->commit();
+    } catch (Exception $e) {
+      $db->rollback();
+    }   
+  }
+
+  public function removeLike()
+  {
+    try {
+      $db = DB::conn();
+      $db->begin();
+      $params = array(
+          $this->id, 
+          $_SESSION['user_id']
+      );
+      $db->query('DELETE FROM likes WHERE
```

Comment:
Same as my comment <a href='https://github.com/adrianneperedo/board/pull/2/files#r26181295'>here</a> po
also in <a href='https://github.com/adrianneperedo/board/pull/2/files#r26181394'>here</a> po, and please apply to all cases :bow: